src/server/adapters/logger/logProvider.ts

```diff
@@ -54,15 +63,73 @@ const getLogDNA = ({ apiKey, environment }: LogDNAParams): LogProvider => {
     };
 };
 
+const getPino = ({ logFilePath, environment }: PinoParams): LogProvider => {
+    const nrPinoConf = nrPino();
+
+    const pinoConf = {
+        ...nrPinoConf,
+        formatters: {
+            ...nrPinoConf.formatters,
+            level: (label: string) => ({ level: label })
+        },
+        base: {
+            service: environment === "prod" ? "microsoft-teams-app" : `microsoft-teams-app-${environment}`
+        }
+    };
+    const stream = pino.destination(logFilePath);
+    /*
+     This is required for pino to keep on writing logfile after log rotation.
+     'SIGHUP' event be triggered by logrotate after log rotation.
+     */
+    process.on("SIGHUP", () => stream.reopen());
+
+    const logger = pino(pinoConf, stream);
+
+    return {
+        info: (message, { meta }): void => logger.info?.({ meta }, message),
+        error: (message, { meta }): void => logger.error?.({ meta }, message),
+        flush: async (): Promise<void> => {
+            logger.flush();
+            await once(logger, "cleared");
+        }
+    };
+};
+
+const getMultipleLogger = ({ apiKey: logDNAApiKey, environment, logFilePath }:MultipleLogParams): LogProvider => {
+    const logdnaLogger = getLogDNA({ apiKey: logDNAApiKey, environment });
+    const pinoLogger = getPino({ environment, logFilePath });
+    return {
+        info: (message:string, { meta }): void => {
+            logdnaLogger.info(message, { meta });
+            pinoLogger.info(message, { meta });
+        },
+        error: (message:string, { meta }): void => {
+            logdnaLogger.error(message, { meta });
+            pinoLogger.error(message, { meta });
+        },
+        flush: async (): Promise<void> => {
+            await logdnaLogger.flush();
+            await pinoLogger.flush();
+        }
+    };
+};
+
 interface LogProviderGetParams {
     logDNAApiKey?: string;
     environment: string;
+    logFilePath: string;
 }
 
-const getLogProvider = ({ logDNAApiKey, environment }: LogProviderGetParams): LogProvider => {
+const getLogProvider = ({ logDNAApiKey, environment, logFilePath }: LogProviderGetParams): LogProvider => {
+    if (logDNAApiKey && logFilePath) {
+        return getMultipleLogger({ apiKey: logDNAApiKey, environment, logFilePath });
+    }
     if (logDNAApiKey) {
         return getLogDNA({ apiKey: logDNAApiKey, environment });
     }
+    if (logFilePath) {
+        return getPino({ logFilePath, environment });
+    }
     return getConsole();
```

Comment:
Do we need this? Pino uses console when there is no logFilePath.
```suggestion
    return getPino({ logFilePath, environment });
```

Reply:
It could be implemented in that way but pino logger always try to open a file stream.

Reply:
pino's console log is in json format and readability is a bit poor, as we already have a formatted console logger, I think it's better to use it.